Source/WebCore/rendering/style/TextUnderlineOffset.h

```diff
@@ -32,60 +32,84 @@ namespace WebCore {
 
 class TextUnderlineOffset {
 public:
-    static constexpr TextUnderlineOffset createWithAuto()
+    static TextUnderlineOffset createWithAuto()
     {
-        return { std::nullopt };
+        return TextUnderlineOffset(Type::Auto);
     }
 
-    static constexpr TextUnderlineOffset createWithLength(float length)
+    static TextUnderlineOffset createWithLength(Length&& length)
     {
-        return { length };
+        return TextUnderlineOffset(Type::Length, WTFMove(length));
     }
 
     constexpr bool isAuto() const
     {
-        return !isLength();
+        return m_type == Type::Auto;
     }
 
     constexpr bool isLength() const
     {
-        return static_cast<bool>(m_length);
+        return m_type == Type::Length;
     }
 
-    void setLengthValue(float length)
+    const Length& length() const
     {
         ASSERT(isLength());
-        m_length = length;
+        return m_length;
     }
 
-    constexpr float lengthValue() const
+    float resolve(float fontSize, float defaultValue = 0.f) const
     {
+        if (isAuto())
+            return defaultValue;
+
         ASSERT(isLength());
-        return *m_length;
+        if (m_length.isPercent())
+            return fontSize * (m_length.percent() / 100.0f);
+        if (m_length.isCalculated())
+            return m_length.nonNanCalculatedValue(fontSize);
+        return m_length.value();
     }
 
-    constexpr float lengthOr(float defaultValue) const
+    constexpr bool operator==(const TextUnderlineOffset& other) const
```

Comment:
We could keep the default generated comparison operator. But it makes me wonder if we want the underlying representation to be a `std::variant<Auto, Length>` actually ?

Reply:
Switched to using std::variant<Auto, Length>. Thanks.